doc/rst/developer/bestPractices/ContributorInfo.rst

```diff
@@ -9,80 +9,297 @@ The git repository for the project is hosted on Github at
+    ``CHPL_COMM=gasnet``.  A common subset is: ``test/release/``,
+    ``test/multilocale/``, and ``test/distributions/``.
+
+  * Changes that are likely to cause portability issues should be tested against
+    different platforms and compilers to avoid fallout in the nightly testing to
+    the extent possible.
+
+* Note that the quickest way to do testing is to use the parallel testing system
+  across a large number of workstations.  If you have limited testing resources
+  available to you, you can request that a member of the core Chapel team help.
+
+.. _Test System: https://github.com/chapel-lang/chapel/blob/master/doc/rst/developer/bestPractices/TestSystem.rst
+
+
+.. _Request feedback on your changes:
+
+Request feedback on your changes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. _Push your work:
+
+Push your work to your feature branch
++++++++++++++++++++++++++++++++++++++
+
+Push your changes to your feature branch on GitHub
+
+.. code-block:: bash
+
+    git push origin <branch_name>
+
+    # or if you don't like typing your complicated branch name,
+    # you can use this command to push the current branch:
+    git push origin HEAD
+
+    # if you forgot your branch name, you can get it by running
+    git branch
+
+    # it is the starred one...
+
+Note that if you have already created a pull request from a
+feature branch, pushing your work to that feature branch will
+update the pull request.
+
+This will enable others to see your work.
+
+.. _Ask for feedback early:
+
+Ask for feedback on your branch early (optional)
+++++++++++++++++++++++++++++++++++++++++++++++++
+
+Not ready to merge yours changes, but still want to see if your work is going
+in the right direction?  Feel free to ask for early feedback!
 
-* Develop your feature, bug fix, etc on your fork. Using a concisely named
-  branch is encouraged.
 
 .. _Submit pull request:
 
 Submit pull request
-~~~~~~~~~~~~~~~~~~~
++++++++++++++++++++
 
 * `Submit a pull request`_ with your changes (make sure you have `synced with
   the main repo`_).
 
-* In order for a pull request to be accepted and merged, it should a) pass
-  testing and b) be reviewed by a member of the core Chapel team (currently, a
-  member of the Cray Chapel team).
+  To do this, after pushing your changes to your feature branch on GitHub,
+  you can use the GitHub web interface to create a pull request. Visit
+
+  ``https://github.com/<username>/chapel``
+
+  and look for a "Compare & pull request" button for your feature branch.
+  Alternatively, navigate to your feature branch, and click the green icon next
+  to the branch dropdown to "Compare, review, create a pull request".
+
+  Next, put in a message to your reviewer about the purpose of your pull request
+  and give the pull request a useful title. It's a good time to draft the commit
+  message that you will need when merging the pull request.  See `Final merge
+  message`_ for recommendations on what that commit message should look like.
```

Comment:
Can we add "It is a good practice to make the PR description match the final merge message. Some developers copy the PR description into the merge message and then wrap any long lines."